src/gurt/hash.c

```diff
@@ -1027,8 +1027,22 @@ d_uhash_link_empty(struct d_ulink *ulink)
 void
 d_hhash_link_insert(struct d_hhash *hhtab, struct d_hlink *hlink, int type)
 {
+	int	n = 0;
+	int	count = 0;
+
 	D_ASSERT(hlink->hl_link.rl_initialized);
 
+	/* check if handle type fits in allocated bits */
+	n = type;
+	while (n) {
+		count++;
+		n >>= 1;
```

Comment:
you can just use max_bits here

1 << max_bits

The new mask should be less than that value

Reply:
I simplified the code like you mentioned in recent commit